src/Networking/NexusMods.Networking.NexusWebApi/NexusModsLibrary.cs

```diff
@@ -71,33 +71,34 @@ public NexusModsLibrary(IServiceProvider serviceProvider)
         var txResults = await tx.Commit();
         return txResults.Remap(newModPage);
     }
-
+    
     /// <summary>
     /// Get or add a collection metadata
     /// </summary>
-    public async Task<CollectionMetadata.ReadOnly> GetOrAddCollectionMetadata(CollectionSlug slug, bool referesh = false, CancellationToken token = default)
+    public async Task<CollectionRevisionMetadata.ReadOnly> GetOrAddCollectionRevision(CollectionSlug slug, RevisionNumber revisionNumber, CancellationToken token)
     {
-        if (!referesh)
-        {
-            var collections = CollectionMetadata.FindBySlug(_connection.Db, slug);
-            if (collections.TryGetFirst(x => x.Slug == slug, out var collection))
-                return collection;
-        }
-
-        var info = await _gqlClient.CollectionInfo.ExecuteAsync(slug.Value, true, token);
-
-        var collectionInfo = info.Data!.Collection;
-        var collectionTileImage = _httpClient.GetByteArrayAsync(new Uri(collectionInfo.TileImage!.ThumbnailUrl), token);
-        var avatarImage = _httpClient.GetByteArrayAsync(new Uri(collectionInfo.User.Avatar), token);
+        var revisions = CollectionRevisionMetadata.FindByRevisionNumber(_connection.Db, revisionNumber)
+            .Where(r => r.Collection.Slug == slug);
+        if (revisions.TryGetFirst(r => r.RevisionNumber == revisionNumber, out var revision)) 
+            return revision;
         
+        var info = await _gqlClient.CollectionRevisionInfo.ExecuteAsync(slug.Value, (int)revisionNumber.Value, true, token);
         using var tx = _connection.BeginTransaction();
+        
         var db = _connection.Db;
+        var collectionInfo = info.Data!.CollectionRevision.Collection;
+        var collectionTileImage = DownloadImage(collectionInfo.TileImage?.ThumbnailUrl, token);
+        var avatarImage = DownloadImage(collectionInfo.User.Avatar, token);
```

Comment:
There should be:

- A way to download the asset at a later date if it is missing.

And ideally also:

- A retry policy for blob (image) downloads.

In periods of high traffic to the site; for example, the Fallout TV release earlier this year, the website can become unresponsive if the load scales past what our regular infrastructure can handle.

In such cases, it's necessary that these assets can be acquired at a later date.
This isn't something that has to be done 'right now' of course, so just open an issue before merging if you'd prefer to do it at a later date.